core/src/main/scala/cats/instances/eq.scala

```diff
@@ -43,4 +45,29 @@ trait EqInstances extends kernel.instances.EqInstances {
       def product[A, B](fa: Eq[A], fb: Eq[B]): Eq[(A, B)] =
         (left, right) => fa.eqv(left._1, right._1) && fb.eqv(left._2, right._2)
     }
+
+  implicit def catsDeferForEq: Defer[Eq] = EqInstances.catsDeferForEqCache
+}
+object EqInstances {
+  private val catsDeferForEqCache: Defer[Eq] =
+    new Defer[Eq] {
+      case class Deferred[A](fa: () => Eq[A]) extends Eq[A] {
```

Comment:
_Just a stylish nitpick that perhaps doesn't affect anything._ I'm not confident about putting underlying classes to the value instantiation.

Reply:
I don't usually use this style, and went with it to remain consistent with the existing instances (like `cats.instances.FunctionInstancesBinCompat0#catsSddDeferForFunction0`).

As I usually avoid it for stylistic reasons, I'm not familiar with the practical concerns around doing it this way, so I'd be interested in hearing why you're worried about it.